track_local_static.go

```diff
@@ -382,6 +439,21 @@ func addExtensions(sample media.Sample, packets []*rtp.Packet) error {
 		}
 	}
 
+	if hyperscaleEncryption {
+		// now check wethear the rest of the packets need to be encrypted
+		for i := 1; i < len(packets); i++ {
+			sampleAttr = 0
+			// set the 'skip encryption' bit
+			if !encryption.ShouldEncrypt(sample, i, payloadDataIdx) {
+				sampleAttr |= 1 << encPosition
+			} else {
+				sampleAttr |= 0 << encPosition
```

Comment:
This is doing nothing. eventually it does OR with 00000000 (This is not a bitmask if you shift a 0)
If encPosition bit was not set before in sampleAttr, it should have already been 0, no need to do anything.
If the bit was one before and you actually want to zero it then OR with 0 bitmask on 1 will do nothing. You should do:
sampleAttr &= ^(1 << encPosition)
(Bit mask 11101111 with &)

Reply:
thanks. since the sampleAtrr is initialized just prior to this i think i'll just remove the else sinse it's initialized with 0 as you say

Reply:
thanks. since the sampleAtrr is initialized just prior to this i think i'll just remove the else sinse it's initialized with 0 as you say